File: app/src/main/java/com/goranch/publicapis/di/PublicApisModule.java

package com.goranch.publicapis.di;

import android.app.Application;

import com.goranch.publicapis.PublicAPIsApp;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PublicApisModule {
    private final PublicAPIsApp app;

    public PublicApisModule(PublicAPIsApp app) {
        this.app = app;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return app;
    }
}

File: app/src/main/java/com/goranch/publicapis/di/PublicApisModule.kt
package com.goranch.publicapis.di

import android.app.Application
import com.goranch.publicapis.PublicAPIsApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PublicApisModule(private val app: PublicAPIsApp) {

    @Provides
    @Singleton
    fun provideApplication(): Application {
        return app
    }
}
